### PR TITLE
Add CI check for any git diff and update bundle.js

### LINKED ISSUE #68: elm-graphql is still outputting Debug.toString when generating scalar debuggers.
I'm trying to compile an app that uses elm-graphql 1.0.0 with elm 0.19 and `--optimize` but it's not compiling as some of my `Api.Object.*` modules have Debug.toString calls in them.

I see that 3720d8d02 was meant to swap these out for the 0.19 equivalents, but if I look in `dist/bundle.js` I can [still see the old Debug.toString implementation](https://github.com/dillonkearns/elm-graphql/blob/master/dist/bundle.js#L7620).  Has something gone wrong with the release?

### PR DESCRIPTION
**Pull request by xtian**
This PR extracts the `approve` npm script into a shell script and changes the `git diff` check to fail if uncommitted changes are produced. This would prevent CI from passing if `dist/bundle.js` is not up to date, for example.

Feel free to close if this isn't how you want it to work! :)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,8 +6,7 @@ node_js:
 
 cache:
   directories:
-    - test/elm-stuff/build-artifacts
-    - node_modules
```
Comment (xtian) [inline .travis.yml]: https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#npm-ci-support

**Comment by dillonkearns** [comment]
Thank you so much @xtian! I love continuously improving this 😄 It's great to have your suggestions and contributions!

> This PR extracts the approve npm script into a shell script and changes the git diff check to fail if uncommitted changes are produced. This would prevent CI from passing if dist/bundle.js is not up to date, for example.

I'm good with that. I still feel like there is something that's not ideal about this workflow for the binaries... I wonder if the ideal solution would have Travis do the release whenever you change the version number in the package.json or something, and then it would automatically generate the binary...? That way the binary wouldn't have to live inside of version control, and there would be no way to forget to generate it, and no need to have a script to check it. Somehow I feel like I'm changing the binary more often than I want to (really, I don't want to ever change it I guess).

But this seems like an improvement. I'll keep mulling over ways to continue to improve it. There's got to be somebody who's figured out a more elegant solution to this problem of compiling binaries for an npm package.

Thanks again for your awesome contributions!

**Comment by dillonkearns** [comment]
It looks like the process for checking in binaries could be removed and made more robust using this: https://docs.travis-ci.com/user/deployment/npm/.

I'm going to take a look at this at some point when I have bandwidth, but for now it's good enough! 😄

**Comment by dillonkearns** [comment]
Alright, I couldn't resist! I'm a sucker for a good improvement. I went ahead and set up travis for automated npm publishing. I tested it out and it's working well! And the npm package and binary are working as expected.

So no more need to commit changes to the `bundle.js` file! 🎉 It's now guaranteed to be using a freshly generated `bundle.js` every time it publishes a new npm version.

08fcc72b1658c6b9c81b2d9f5de03bec3b59393b is the relevant commit.

Here's what the output looks like when it does a release: https://travis-ci.org/dillonkearns/elm-graphql/builds/421806163

I should probably clean up the `.npmignore` file so the npm package assets aren't so big. But I'll save that optimization for another day.

**Comment by xtian** [comment]
@dillonkearns Nice, that's great! Much better way to do it

**Pull request merged**